### ISSUE TITLE
market_benchmark.py yields very poor results on FastRT

### ISSUE THREAD

**Issue by jamitr-jci** [JDAI-CV/fast-reid#658]
## Instructions To Reproduce the Issue:

- Installed using the provided docker image inside `projects/FastRT/pybind_interface/docker/trt7cu102_torch160/`
- Converted `sbs_R50-ibn.wts` to `sbs_R50-ibn.engine` 
- Run `python3 market_benchmark.py` 
- Results are very poor

1. full code you wrote or full changes you made (`git diff`)
```
diff --git a/projects/FastRT/pybind_interface/market_benchmark.py b/projects/FastRT/pybind_interface/market_benchmark.py
index 51c9caf..5c8404e 100755
--- a/projects/FastRT/pybind_interface/market_benchmark.py
+++ b/projects/FastRT/pybind_interface/market_benchmark.py
@@ -29,7 +29,7 @@ GPU_ID = 0
 def map(wrapper):
        model = wrapper
        cfg = get_cfg()
-       test_loader, num_query = build_reid_test_loader(cfg, "Market1501", T.Compose([]))
+       test_loader, num_query = build_reid_test_loader(cfg, dataset_name="Market1501", transforms=T.Compose([]))

        feats = []
        pids = []
@@ -40,8 +40,8 @@ def map(wrapper):
                        t = torch.Tensor(np.array([model.infer(cv2.imread(image_path))]))
                        t.to(torch.device(GPU_ID))
                        feats.append(t)
-               pids.extend(batch["targets"].numpy())
-               camids.extend(batch["camids"].numpy())
+               pids.extend(batch["targets"].cpu().numpy())
+               camids.extend(batch["camids"].cpu().numpy())

        feats = torch.cat(feats, dim=0)
        q_feat = feats[:num_query]

```
2. what exact command you run: `python3 market_benchmark.py` 
3. __full logs__ you observed:
```
python3 market_benchmark.py
../../../fastreid/evaluation/rank.py:15: UserWarning: Cython rank evaluation (very fast so highly recommended) is unavailable, now use python evaluation.
  'Cython rank evaluation (very fast so highly recommended) is '
Init on device 0
[05/02/2022-11:50:09] [W] [TRT] Current optimization profile is: 0. Please ensure there are no enqueued operations pending in this context prior to switching profiles
../../../fastreid/data/datasets/market1501.py:45: UserWarning: The current data structure is deprecated. Please put data folders such as "bounding_box_train" under "Market-1501-v15.09.15".
  warnings.warn('The current data structure is deprecated. Please '
mAP 0.006028478418972129, rank-1 0.003562945406883955
Destroy engine succeed

```

## Expected behavior:

Observed considerable accuracy drop converting provided trained model to TRT. Using the pretrained `sbs_R50-ibn.pth` and the original scripts I got very poor results on the converted engine. Had to slightly change the benchmark scripts due to a possible bug.

**Comment by caoshuaibingcao** [comment]
me too;请问你解决了吗？

**Comment by jamitr-jci** [comment]
@caoshuaibingcao I compiled without the docker (fresh TRT install etc.) and got the exact same result.

**Comment by jamitr-jci** [comment]
Solved the issue by normalizing the feature vectors. Matrix multiplication cannot be done on unnormalized features (it is not equal to cosine distance).
```
q_feat = feats[:num_query]
q_feat = torch.nn.functional.normalize(q_feat, dim=1) # Normalize
g_feat = feats[num_query:]
g_feat = torch.nn.functional.normalize(g_feat, dim=1) # Normalize
...
distmat = 1 - torch.mm(q_feat, g_feat.t()) # this line works correctly only on normalized features
```
